examples/jsm/renderers/webgpu/nodes/WGSLNodeBuilder.js

```diff
@@ -208,12 +208,16 @@ class WGSLNodeBuilder extends NodeBuilder {
 
 	}
 
-	generateTextureLoad( texture, textureProperty, uvIndexSnippet, depthSnippet, levelSnippet = '0u' ) {
+	generateTextureLoad( texture, textureProperty, uvIndexSnippet, depthSnippet, levelSnippet = '0u', accessSnippet ) {
 
 		if ( depthSnippet ) {
 
 			return `textureLoad( ${ textureProperty }, ${ uvIndexSnippet }, ${ depthSnippet }, ${ levelSnippet } )`;
 
+		} else if ( accessSnippet ) {
```

Comment:
Could we detect if it's a `StorageTexture` here instead of moving the setAccess() to TextureNode?

Reply:
Technically since `textureLoad` inherits from `TextureNode` and not `StorageTextureNode` it's not possible.
We need to be able to specify the access in the reading (`textureLoad`) via `textureLoad( readTex, ivec2( x, y ) ).setAccess( 'read-only' )` that can differ from the writing `textureStore( writeTex, uv, value, 'read-write' )`.

Alternative would be to add a 4th argument for textureLoad:
`textureLoad( readTex, ivec2( x, y ), null, 'read-only' )`

Reply:
I think we would extend `generateSnippet` to support `generateTextureLoad` with access on `StorageTextureNode` and keep `TextureNode` as it was. This was part of the implementation although not finalized use `textureStore` just to read texture if you don't add `value` as parameter to assign, for example: `textureStore( storageTexture, indexUV )`